### PR TITLE
CWMSVue 594 - Added location office parameter to LocationGroup getAll

### PR DESCRIPTION
**Pull request by zack-rma**
Added location office parameter to LocationGroup getAll

### CODE REVIEW

File: cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java [hunk 1]
```diff
@@ -273,10 +279,21 @@ public List<LocationGroup> getLocationGroups(String groupOfficeId, String catego
                 .on(alcg.LOC_CATEGORY_ID.eq(alga.CATEGORY_ID)
                         .and(alcg.LOC_GROUP_ID.eq(alga.GROUP_ID)));
 
-        if (groupOfficeId != null) {
-            connectBy = onStep.where(DSL.upper(alcg.GRP_DB_OFFICE_ID).eq(groupOfficeId.toUpperCase()).and(condition));
+        if (officeId != null) {
```
Comment (adamkorynta) [inline cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java]: Seems like these conditions can be simplified. If group office id is not null, then filter group. Else if location office is not null then filter location office. Not sure why there are four branches here

File: cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java [hunk 2]
```diff
@@ -352,18 +370,24 @@ public List<LocationGroup> getLocationGroups(String groupOfficeId,
             condition = condition.and(alcg.CAT_DB_OFFICE_ID.eq(categoryOfficeId.toUpperCase()));
         }
 
-        if (groupOfficeId != null) {
-            if (CWMS.equalsIgnoreCase(groupOfficeId)) {
+        if (officeId != null) {
```
Comment (adamkorynta) [inline cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java]: for consistent and clarified naming, use "locationOfficeId" (including naming in controller and other methods)

File: cwms-data-api/src/main/java/cwms/cda/api/LocationGroupController.java [hunk 3]
```diff
@@ -76,9 +76,10 @@ private Timer.Context markAndTime(String subject) {
         return Controllers.markAndTime(metrics, getClass().getName(), subject);
     }
 
-    @OpenApi(queryParams = {
+    @OpenApi(
+        queryParams = {
             @OpenApiParam(name = OFFICE, description = "Specifies the owning office of the "
```
Comment (adamkorynta) [inline cwms-data-api/src/main/java/cwms/cda/api/LocationGroupController.java:82]: Changing the usage of a parameter breaks backwards compatibility

File: cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java (hunk 1 above)
Comment (zack-rma) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java]: Simplified conditioning

File: cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java (hunk 2 above)
Comment (zack-rma) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java]: renamed where appropriate

File: cwms-data-api/src/main/java/cwms/cda/api/LocationGroupController.java (hunk 3 above)
Comment (zack-rma) [reply cwms-data-api/src/main/java/cwms/cda/api/LocationGroupController.java:82]: reverted functionality and moved location office to additional query parameter

**Review by adamkorynta — APPROVED** [review]

**Pull request merged**